src/plugins/ethereum/ethAddress.js

```diff
@@ -0,0 +1,39 @@
+// @flow
+
+import {isAddress, toChecksumAddress} from "web3-utils";
+
+import * as C from "../../util/combo";
+import {compatibleParser} from "../../util/compat";
+
+export opaque type EthAddress: string = string;
+
+export function parseAddress(s: string): EthAddress {
+  if (!isAddress(s)) {
+    throw new Error(`not a valid ethereum address: ${s}`);
+  }
+  return toChecksumAddress(s);
+}
+
+// Utilized for a more readable and recognizable address. This is
+// needed to allow the address to fit within the Identity name requirements
+export function truncateEthAddress(address: EthAddress): string {
+  const prefix = address.slice(0, 6);
+  const suffix = address.slice(-4);
+  return `${prefix}...${suffix}`;
+}
+
+export const ethAddressParser: C.Parser<EthAddress> = C.fmap(
+  C.string,
+  parseAddress
+);
+
+export const COMPAT_INFO = {
+  type: "sourcecred/ethAddressFile",
+  version: "0.0.1",
+};
+
+const addressEntriesParser = C.array(ethAddressParser);
+
+export const parser: C.Parser<
+  Array<EthAddress>
+> = compatibleParser(COMPAT_INFO.type, {"0.0.1": addressEntriesParser});
```

Comment:
are we using JSON log, or a file with an array of addresses? Confused based on the PR description.

Reply:
File with array of addresses: see #2420. I like this format because it
keeps the `Compatible` structure without paying a 50-byte penalty on
every 40-byte line, and also keeps the format more easily extensible to
other kinds of configuration, which seem plausibly desirable.

Do clarify PR description